sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```diff
@@ -299,213 +303,152 @@ public void saveDistroPropertiesTo(File destination, String distro) throws MojoE
+			if (moduleDir.exists() && moduleDir.isDirectory()) {
+				for (File file : Objects.requireNonNull(moduleDir.listFiles())) {
+					if (file.isDirectory()) {
+						String[] fileComponents = file.getName().split("-");
+						StringBuilder moduleName = new StringBuilder();
+						StringBuilder version = new StringBuilder();
+						for (int i = 1; i < fileComponents.length; i++) {
+							String component = fileComponents[i];
+							if (Semver.isValid(component) || version.length() > 0) {
+								if (version.length() > 0) {
+									version.append("-");
+								}
+								version.append(component);
+							} else {
+								if (moduleName.length() == 0) {
+									moduleName.append("@").append(fileComponents[0]).append("/");
+								}
+								else {
+									moduleName.append("-");
+								}
+								moduleName.append(component);
+							}
+						}
+						ret.put(moduleName.toString(), version.toString());
 					}
-
-					distroProperties.addProperty(dependency, latestVersion);
-				} else {
-					checkVersionInRange(dependency, versionRange, distroVersion, contentProperties.getProperty("name"));
 				}
 			}
 		}
+		return ret;
 	}
 
 	/**
-	 * Checks if the version from distro.properties satisfies the range specified in content.properties.
-	 * Throws an exception if there is a mismatch.
-	 *
-	 * @param contentDependencyKey The key of the content dependency.
-	 * @param contentDependencyVersionRange The version range specified in content.properties.
-	 * @param distroPropertyVersion The version specified in distro.properties.
-	 * @param contentPackageName The name of the content package.
-	 * @throws MojoExecutionException If the version does not fall within the specified range or if the
-	 *             range format is invalid.
+	 * Alternative version of the `satisfies` functionality from semver4j that allows snapshot and pre-releases from
+	 * later versions to satisfy earlier release ranges
+	 * This also will always return false if the passed version is blank or null, and always return true if the version is "next"
 	 */
-	private static void checkVersionInRange(String contentDependencyKey, String contentDependencyVersionRange, String distroPropertyVersion, String contentPackageName) throws MojoExecutionException {
-		Semver semverVersion = new Semver(distroPropertyVersion);
-
-		try {
-			boolean inRange = semverVersion.satisfies(contentDependencyVersionRange.trim());
-			if (!inRange) {
-				throw new MojoExecutionException("Incompatible version for " + contentDependencyKey + " in content package "
-						+ contentPackageName + ". Specified range: " + contentDependencyVersionRange
-						+ ", found in distribution: " + distroPropertyVersion);
-			}
-		} catch (SemverException e) {
-			throw new MojoExecutionException("Invalid version range format for " + contentDependencyKey
-					+ " in content package " + contentPackageName + ": " + contentDependencyVersionRange, e);
```

Comment:
Per slack discussion this new method is what checks for version incompatibility, accounting for things like "next", and also allowing snapshot and pre-releases that are otherwise higher than a release version to satisfy that release version.  See unit test in `DistroHelperTest` for examples.